plugins/transport-grpc/src/main/java/org/opensearch/transport/grpc/ssl/SecureAuxTransportSslContext.java

```diff
@@ -0,0 +1,194 @@
+        switch (clientAuthStr) {
+            case "NONE" -> {
+                return NONE;
+            }
+            case "OPTIONAL" -> {
+                return OPTIONAL;
+            }
+            case "REQUIRE" -> {
+                return REQUIRE;
+            }
+            default -> throw new OpenSearchSecurityException("unsupported client auth: " + clientAuthStr);
+        }
+    }
+
+    /**
+     * Simple ssl provider string to enum conversion helper.
+     * @param providerStr provider as string.
+     * @return provider enum.
+     */
+    public static SslProvider providerHelper(String providerStr) {
+        switch (providerStr) {
+            case "JDK" -> {
+                return JDK;
+            }
+            case "OPENSSL" -> {
+                return OPENSSL;
+            }
+            case "OPENSSL_REFCNT" -> {
+                return OPENSSL_REFCNT;
+            }
+            default -> throw new OpenSearchSecurityException("unsupported ssl provider: " + providerStr);
+        }
+    }
+
+    /**
+     * Initializes a new SecureAuxTransportSslContext.
+     * @param provider source of SecureAuxTransportParameters required to build an SslContext.
+     */
+    public SecureAuxTransportSslContext(SecureAuxTransportSettingsProvider provider) {
+        Optional<SecureAuxTransportSettingsProvider.SecureAuxTransportParameters> params = provider.parameters();
+        if (params.isEmpty()) {
+            throw new OpenSearchSecurityException("Aux transport ssl context failed to initialize. Secure settings provider not found.");
+        }
+        try {
+            this.sslContext = buildContext(params.get());
+        } catch (SSLException e) {
+            throw new OpenSearchSecurityException("Unable to build io.grpc.SslContext from secure settings", e);
+        }
+    }
+
+    /**
+     * @param p fields necessary to construct an SslContext.
+     * @return new SslContext.
+     */
+    private SslContext buildContext(SecureAuxTransportSettingsProvider.SecureAuxTransportParameters p) throws SSLException {
+        if (p.keyManagerFactory().isEmpty()) {
+            throw new OpenSearchSecurityException("Aux transport ssl context failed to initialize. No keystore provided.");
+        }
+        if (p.sslProvider().isEmpty()) {
+            throw new OpenSearchSecurityException("Aux transport ssl context failed to initialize. Ssl provider not found.");
+        }
+        if (p.clientAuth().isEmpty()) {
+            throw new OpenSearchSecurityException("Aux transport ssl context failed to initialize. No client auth mode configured.");
+        }
+        if (p.trustManagerFactory().isEmpty()) {
+            throw new OpenSearchSecurityException("Aux transport ssl context failed to initialize. No truststore provided.");
+        }
+        SslContextBuilder builder = SslContextBuilder.forServer(p.keyManagerFactory().get())
+            .sslProvider(providerHelper(p.sslProvider().get()))
```

Comment:
wdyt about using the netty enum: `io.netty.handler.ssl.SslProvider.valueOf(p.sslProvider.get())`?